### PR TITLE
fix status updates on dims changes

### LINKED ISSUE #286: Changing dims slider no longer updates status message
## 🐛 Bug

Changing the dims slider no longer updates status message in the bottom left corner

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR fixes #286 by causing the status message to update when the dims slider changes. It does this for the `Image`, `Labels`, and `Markers` layers. The `Shapes` and `Vectors` layers do not currently support > 2D so it is not needed there yet, but will be implemented when needed.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] run `examples/nD_image.py` and `examples/nD_markers.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sofroniewn** [comment]
There was an issue with multichannel data that was causing an error on mouse move on master, but with this PR because an error on launch of `set_themes.py` because it uses an rgb image, that has now been fixed.

**Comment by kne42** [comment]
it's failing [here](https://github.com/napari/napari/blob/d2e43eb29fba659d520167be0c1615b7f29b7c77/napari/components/_dims/view.py#L73). my first guess would be that it probably incurred circular callbacks

**Comment by sofroniewn** [comment]
@kne42 this PR is passing now, I think we've got wires crossed on our two PRs

File: napari/layers/_base_layer/model.py [hunk 1]
```diff
@@ -119,14 +121,18 @@ def coordinates(self):
         The setter expects the a 2-tuple of coordinates in canvas space
         ordered (x, y) and then transforms them to image space and inserts
         them into the correct position of the layer indices. The length of the
-        tuple is equal to the number of dimensions of the layer.
+        tuple is equal to the number of dimensions of the layer. If `None` is
```
Comment (kne42) [inline napari/layers/_base_layer/model.py]:
```suggestion
        tuple is equal to the number of dimensions of the layer. If ``None`` is
```

File: napari/layers/_base_layer/model.py [hunk 2]
```diff
@@ -119,14 +121,18 @@ def coordinates(self):
...
+        passed then (0, 0) in will replace necessary the layer indices.
```
Comment (kne42) [inline napari/layers/_base_layer/model.py]: the "then (0, 0) in" confuses me, is this a typo or am i just bad at english?

**Review by kne42 — APPROVED** [review]

File: napari/layers/_base_layer/model.py (hunk 2 above)
Comment (sofroniewn) [reply napari/layers/_base_layer/model.py]: type - "in" should not be there!

Comment (sofroniewn) [reply napari/layers/_base_layer/model.py]: *typo

**Pull request merged**